proposals/xxxx-package-manager-local-dependencies.md

```diff
@@ -0,0 +1,72 @@
+# Package Manager Local Dependencies
+
+* Proposal: [SE-XXXX](XXXX-package-manager-local-dependencies.md)
+* Authors: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: [TBD](https://github.com/)
+* Status: **Discussion**
+
+## Introduction
+
+This proposal adds support for depending on a local package in the manifest
+file.
+
+## Motivation
+
+Edit mode allows developing on set of packages without first committing and
+tagging the changes in a dependency. However, that is a local state which can't
+be shared within a development team and is not persisted if the build directory
+is reset.
+
+Moreover, there is an unnecessary friction when a package author is creating
+a set of packages because the packages need to be under source control before
+edit mode can be used.
+
+## Proposed solution
+
+We propose to add a method in `Package.Dependency` to allow specifying local dependencies.
+
+```swift
+/// Adds dependency to a package with a given local path.
+public static func package(localPath: String) -> Package.Dependency
+```
+
+Example:
+
+```swift
+// swift-tools-version:4.0.0
+
+import PackageDescription
+
+let package = Package(
+    name: "Foo",
+    dependencies: [
+        .package(localPath: "bar/corge"),
+        .package(localPath: "../baz"),
+        .package(localPath: "/path/to/qux"),
+    ]
+)
+```
+
+## Detailed design
+
+The local dependencies are not required to be placed under source control and
+its sources will be used as-is. There are no restrictions on the path and it can
+be relative or absolute.
+
+All local dependencies should be in the root package or reachable via other
```

Comment:
I think one obvious question that will come up for people is the case of wanting multiple packages within the same repo, in which case they might want to have the top-level package in the repo have a local reference to a "wholly owned" package that is at some subpath inside the repo.  Since we're not intended to start the multi-package repo discussion here, we should make explicit that this is the reason we're disallowing local path references inside remote package manifests.